diff --git a/small_programs/my_menu/menu_e.c b/small_programs/my_menu/menu_e.c
index ec189f2..65fb362 100755
--- a/small_programs/my_menu/menu_e.c
+++ b/small_programs/my_menu/menu_e.c
@@ -1,29 +1,33 @@
-#include <stdio.h>
-#include <ctype.h>
-
-/* Excercise 2
-Petle while, do while lub for
-Cykl powtarza sie tak dlugo dopoki uzytkownik nie wcisnie Entera /nowej linii
-enter opuszcza bufor na samym koncu>> ?
-jesli wpisze sie kilka znakow abcfdjksdf to one opuszcza (wszystkie) bufor wraz z enterem
-program musi je odsylac lub polykac.
-*/
-
-int main() {
-	char znak;
-	printf("\nProsze wprowadzic znak:");
-	znak = getchar();
-	getchar();
-	if (isdigit(znak)) {
-	 		printf("Wpisales cyfre: %c.", znak);
-	   		printf("\nDla ASCII jest to %d.", znak);
- 	} else if (isalpha(znak)) {
-    		printf("\nWpisales litere: %c", znak);
- 	} else {
-	   		printf("\nWpisales inny znak: %c", znak);
-	   		getchar();
-  	}
-	while('\n' != getchar());
-	return 0;
-}
-
+#include <stdio.h>
+#include <ctype.h>
+
+/* Excercise 2
+Petle while, do while lub for
+Cykl powtarza sie tak dlugo dopoki uzytkownik nie wcisnie Entera /nowej linii
+enter opuszcza bufor na samym koncu>> ?
+jesli wpisze sie kilka znakow abcfdjksdf to one opuszcza (wszystkie) bufor wraz z enterem
+program musi je odsylac lub polykac.
+*/
+
+int main() {
+	char znak;
+	while(1){
+		printf("\nProsze wprowadzic znak:");
+		znak = getchar();
+		if(znak == '\n')
+			break;
+		if (isdigit(znak)) {
+			printf("Wpisales cyfre: %c.", znak);
+	   		printf("\nDla ASCII jest to %d.", znak);
+ 		} else if (isalpha(znak)) {
+    			printf("\nWpisales litere: %c", znak);
+ 		} else {
+			printf("\nWpisales inny znak: %c", znak);
+  		}
+	
+		while('\n' != getchar());
+	}
+
+	return 0;
+}
+
